src/components/ui/NavBar/index.tsx

```diff
@@ -43,12 +46,16 @@ const NavBar = () => {
         }
       });
     };
-
     window.addEventListener("scroll", handleScroll);
     handleScroll();
+    if (pathname === "/") {
+      setActiveSection("home");
+    } else {
+      setActiveSection(pathname);
+    }
```

Comment:
Optional: Can we try to use ternary operators here? It will reduce lines of code. Just suggestion
```suggestion
    setActiveSection(pathname === "/" ? "home" : pathname);

```